diff -up openjdk/src/hotspot/share/code/compiledIC.cpp.omv~ openjdk/src/hotspot/share/code/compiledIC.cpp
--- openjdk/src/hotspot/share/code/compiledIC.cpp.omv~	2019-01-15 20:45:43.990205590 +0100
+++ openjdk/src/hotspot/share/code/compiledIC.cpp	2019-01-15 20:45:57.843364512 +0100
@@ -224,7 +224,7 @@ bool CompiledIC::set_to_megamorphic(Call
     assert(bytecode == Bytecodes::_invokeinterface, "");
     int itable_index = call_info->itable_index();
     entry = VtableStubs::find_itable_stub(itable_index);
-    if (entry == false) {
+    if (!entry) {
       return false;
     }
 #ifdef ASSERT
